processing/src/main/java/org/apache/carbondata/processing/sortandgroupby/sortdata/SortDataRows.java

```diff
@@ -193,20 +193,42 @@ public void startSorting() throws CarbonSortKeyAndGroupByException {
       toSort = new Object[entryCount][];
       System.arraycopy(recordHolderList, 0, toSort, 0, entryCount);
 
-      if (parameters.isUseKettle()) {
-        if (parameters.getNoDictionaryCount() > 0) {
-          Arrays.sort(toSort, new RowComparator(parameters.getNoDictionaryDimnesionColumn(),
-              parameters.getNoDictionaryCount()));
+      if (parameters.getNumberOfSortColumns() == 0) {
```

Comment:
If sort columns are none, then you are not supposed to sort it right? And also there are lot of checks it supposed to be simplified.